tests/sentry/utils/appleconnect/test_appstore_connect.py

```diff
@@ -1,9 +1,169 @@
+"""Tests for using the App Store Connect API.
+
+Some of these tests will try to use the live API requests if you provide a ``apikey.json``
+file with credentials.  See the ``api_credentials`` fixture for details.
+"""
+
+import pathlib
+import textwrap
+import urllib.parse
+
 import pytest
+import requests
+import responses as responses_mod
 
 from sentry.lang.native.appconnect import NoDsymUrl
+from sentry.utils import json
 from sentry.utils.appleconnect import appstore_connect
 
 
+class TestListBuilds:
+    @pytest.fixture(scope="session", params=["live", "responses"])
+    def api_credentials(self, request) -> appstore_connect.AppConnectCredentials:
+        """An App Store Connect API key in the form of AppConnectCredentials.
+
+        If ``apikey.json`` is present in the current directory it will load the credentials
+        from this json for the ``live`` param, the format should look like this:
+
+        ```json
+            {
+                "key_id": "AAAAAAAAAA",
+                "issuer_id": "abcdef01-2345-6789-abcd-ef0123456789",
+                "private_key": "-----BEGIN PRIVATE KEY-----\na_bunch_of_\n_separated_base64\n-----END PRIVATE KEY-----\n"
+            }
+        ```
+
+        For the ``responses`` param a fake value is returned with a key_id="AAAAAAAAAA".
+        """
+        if request.param == "live":
+            here = pathlib.Path(__file__).parent
+            keyfile = here / "apikey.json"
+            try:
+                data = json.loads(keyfile.read_text(encoding="utf-8"))
+            except FileNotFoundError:
+                pytest.skip("No API key available for live tests")
+            else:
+                return appstore_connect.AppConnectCredentials(
+                    key_id=data["key_id"], issuer_id=data["issuer_id"], key=data["private_key"]
+                )
+        else:
+            # NOTE: This key has been generated for these test.
```

Comment:
what does this mean exactly? where are the contents of the not-live (dead? undead, unlive?) private key sourced from?

Reply:
I mean the key has been freshly generated and is just a random key, it will not be recognised anywhere and is thus safe to commit to our source code.

I could entirely remove the comment if you prefer.

Reply:
thanks for the explanation! i think keeping the comment is good although i'd love to see it rephrased to something more along the lines of the first sentence of your reply, e.g.

```py
# NOTE: This key has been generated outside of App Store Connect for the sole purpose of being used in this test. The key is not associated with any account on App Store Connect, and therefore is unable to access any live data.
```

the original comment leaves a lot of details out, namely that it's not associated with anything and therefore is essentially a throwaway key that can't do anything dangerous.